sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -2776,7 +2776,11 @@ object functions {
    * @since 3.3.0
    */
   def lpad(str: Column, len: Int, pad: Array[Byte]): Column = withExpr {
-    BinaryPad("lpad", str.expr, lit(len).expr, lit(pad).expr)
```

Comment:
let's just create `UnresolvedFunction` here.

Reply:
Do you mean when the legacy is restored, and `pad` is binary? Why is it a `UnresolvedFunction`?

Reply:
By using `UnresolveFunction`, we will go through the function look up code path and your change in `PadExpressionBuilderBase` will kick in.

Reply:
Oh thanks, never know that.